### PR TITLE
[Jira] Fix NotFound when the path is present in host URL

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Closes https://github.com/elastic/connectors/issues/2343

This PR removes leading slashes from the endpoints used in the connector and adding an extra '/' in the host url to prevent the NotFound error.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

## Release Notes
Adds support for Jira Server and Jira Data Center where the root of the server is hosted behind a path, and not at a root of the host/domain.

### CODE REVIEW

File: connectors/sources/jira.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def __init__(self, configuration):
         self.configuration = configuration
         self._logger = logger
         self.data_source_type = self.configuration["data_source"]
-        self.host_url = self.configuration["jira_url"]
+        self.host_url = self.configuration["jira_url"] + "/"
```
Comment (artem-shelkovnikov) [inline connectors/sources/jira.py:107]: Should we prune `/` in the end of the `self.configuration["jira_url"]` if present?

Comment (moxarth-elastic) [reply connectors/sources/jira.py:107]: `urljoin` will take care of it if the host url contains double slash at the end. However, we can add it only if the trailing slash is not present.

File: tests/sources/test_jira.py [hunk 2]
```diff
@@ -555,6 +559,15 @@ async def test_get_with_500_status():
                     await response.json()
 
 
+@pytest.mark.asyncio
+async def test_ping_to_custom_path_server():
+    async with create_jira_source(jira_url="http://127.0.0.1:8080/test/") as source:
+        with patch.object(
+            aiohttp.ClientSession, "get", side_effect=side_effect_function
+        ):
+            await source.ping()
```
Comment (seanstory) [inline tests/sources/test_jira.py:568]: let's add an assertion here that checks that the pinged url is prefixed with `/test/`

**Review by seanstory — COMMENTED** [review]
one nit, but looks straightforward

**Comment by seanstory** [comment]
@moxarth-elastic I added the `release-note` label and a section to the issue description. Don't forget to add these for bugfixes so that it's clear what changes from release to release. :)

**Comment by moxarth-elastic** [comment]
> @moxarth-elastic I added the `release-note` label and a section to the issue description. Don't forget to add these for bugfixes so that it's clear what changes from release to release. :)

Got it !

**Comment by seanstory** [comment]
@moxarth-elastic can you be sure to manually create the backport to address the conflicts? ☝️

**Comment by seanstory** [comment]
woops, @praveen-elastic , not moxarth. Sorry!

**Comment by moxarth-elastic** [comment]
> @moxarth-elastic can you be sure to manually create the backport to address the conflicts? ☝️

Raised a Backport PR: https://github.com/elastic/connectors/pull/2388

**Pull request merged**